hibernate-core/src/test/java/org/hibernate/test/util/StringHelperTest.java

```diff
@@ -71,6 +71,8 @@ public void testArrayUnquoting() {
 		unchanged( new String[] { "a", "b" } );
 		helperEquals( new String[] { "a", "b", "c" }, new String[] { "a", "b", "`c`" } );
 		helperEquals( new String[] { "a", "b", "c" }, new String[] { "a", "\"b\"", "c" } );
+		helperEquals( new String[] { "\"a", "b\"", "`c", "d`" }, new String[] { "\"a", "b\"", "`c", "d`" } );
+		helperEquals( new String[] { "\"a`", "`b\"", "`c\"", "\"d`" }, new String[] { "\"a`", "`b\"", "`c\"", "\"d`" } );
```

Comment:
why should we test for these? what value does it add?

I know this is how it works today but I'm ok that it's unspecified, no?